File: apps/react-native/src/models/Patient.js
/**
 * Patient Model and Criticality Enum
 * Equivalent to Flutter's Patient class and PatientCriticality enum
 */

export const PatientCriticality = {
  CRITICAL: 'critical',
  HIGH: 'high',
  MEDIUM: 'medium',
  LOW: 'low',
};

export class Patient {
  constructor(firstName, lastName, criticality = null, nextVisit = null) {
    this.firstName = firstName;
    this.lastName = lastName;
    this.criticality = criticality;
    // Ensure nextVisit is always a Date object, not a string
    this.nextVisit = nextVisit ? (nextVisit instanceof Date ? nextVisit : new Date(nextVisit)) : null;
  }

  get fullName() {
    return `${this.firstName} ${this.lastName}`;
  }
}

File: apps/flutter/lib/core/patients/patient.dart
enum PatientCriticality { critical, high, medium, low }

class Patient {
  final String firstName;
  final String lastName;

  /// If null: not in “needing attention”
  final PatientCriticality? criticality;

  /// If null: no upcoming visit
  final DateTime? nextVisit;

  const Patient({
    required this.firstName,
    required this.lastName,
    this.criticality,
    this.nextVisit,
  });

  String get fullName => '$firstName $lastName';
}
